### PR TITLE
Use 'skipInterface: ^br-.*' for Docker Enterprise ipv4 autodetection …

### PR DESCRIPTION
**Pull request by Brian-McM**
…method

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/installation/core_controller.go [hunk 1]
```diff
@@ -455,10 +455,18 @@ func fillDefaults(instance *operator.Installation) error {
...
 			if instance.Spec.CalicoNetwork.NodeAddressAutodetectionV4 == nil {
-				// Default IPv4 address detection to "first found" if not specified.
-				t := true
-				instance.Spec.CalicoNetwork.NodeAddressAutodetectionV4 = &operator.NodeAddressAutodetection{
-					FirstFound: &t,
+				if instance.Spec.KubernetesProvider == operator.ProviderDockerEE {
+					// firstFound finds the Docker Enterprise interface prefixed with br-, which is unusable for the
+					// node address, so instead skip the interface br-.
+					instance.Spec.CalicoNetwork.NodeAddressAutodetectionV4 = &operator.NodeAddressAutodetection{
+						SkipInterface: "^br-.*",
```
Comment (caseydavenport) [inline pkg/controller/installation/core_controller.go:462]: I worry that we also need to skip a lot of other interfaces as well.

I think we still need a regex that also matches these: https://github.com/projectcalico/node/blob/904c449d84f3d9e0bcfde849f3076869e1a2a4ec/pkg/startup/startup_linux.go#L29-L33

WDYT? I'm not 100% up to speed on Docker enterprise - will it ever have any of those other interfaces on it as well?

Comment (Brian-McM) [reply pkg/controller/installation/core_controller.go:462]: That's a really good point.... I'll double check if it's needed and add it if it is.

Comment (Brian-McM) [reply pkg/controller/installation/core_controller.go:462]: @caseydavenport https://github.com/projectcalico/node/blob/master/pkg/startup/startup.go#L815-L828, looks like we still use those interfaces to skip.

**Review by matthewdupre — APPROVED** [review]

File: pkg/controller/installation/core_controller.go (hunk 1 above)
Comment (caseydavenport) [reply pkg/controller/installation/core_controller.go:462]: Awesome, :shipit:

**Pull request merged**